server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java

```diff
@@ -108,27 +108,28 @@ protected Query literalMatchesAnyArrayRef(Function any, Literal<?> probe, Refere
         Object value = probe.value();
         BooleanQuery.Builder query = new BooleanQuery.Builder();
         query.setMinimumNumberShouldMatch(1);
-        query.add(
-            eqQuery.rangeQuery(
-                columnName, value,
-                null,
-                false,
-                false,
-                candidates.hasDocValues(),
-                candidates.indexType() != IndexType.NONE),
-            BooleanClause.Occur.SHOULD
-        );
-        query.add(
-            eqQuery.rangeQuery(
-                columnName,
-                null,
-                value,
-                false,
-                false,
-                candidates.hasDocValues(),
-                candidates.indexType() != IndexType.NONE),
-            BooleanClause.Occur.SHOULD
-        );
+        var gt = eqQuery.rangeQuery(
+            columnName,
+            value,
+            null,
+            false,
+            false,
+            candidates.hasDocValues(),
+            candidates.indexType() != IndexType.NONE);
+        var lt = eqQuery.rangeQuery(
+            columnName,
+            null,
+            value,
+            false,
+            false,
+            candidates.hasDocValues(),
+            candidates.indexType() != IndexType.NONE);
+        assert lt != null || gt == null : "If lt is null, gt must be null";
```

Comment:
This assertion should inside the if below with the `&&`, before `return null` like this:
```
assert lt != null || gt != null : "Both gt and lt should be null";
```

Reply:
Wouldn't it be already `lt == null` such that `assert lt != null ...` be redundant?

Reply:
yep, sorry, the `if` should also become an `||`, so you get in if one of the 2 is is null, then you assert that both should be null, and then `return null`.